www/source/blog/2017-10-24-node-with-db.html.md

```diff
@@ -0,0 +1,348 @@
+
+Now that we have this defined, we can use this config:
+
+```
+var dbConnectionString = 'mongodb://' + nconf.get('mongo')['host'] + ':' + nconf.get('mongo')['port'] + "/test'";
+```
+
+Here, we create a db connection string - pulling the ('mongo')['host'] and ('mongo')['port'] values from the configuration file we just loaded.
+
+In this case, we uses the default database included with all new installs of MongoDB - the test database - at the end of the db connection string.
+
+```
+var mongoose = require('mongoose');
+mongoose.connect(dbConnectionString, { useMongoClient: true });
+```
+
+Finally, we use the mongoose node module to connect to the database. Go ahead and save and close the file.
+
+Open up your package.json file and add nconf as a dependency:
+
+**package.json**
+
+```
+  "dependencies": {
+    (...)
+	"nconf": "^0.8.5",
+    (...)
+  }
+```
+
+And then re-run npm install
+
+```console
+$ npm install
+```
+
+One more thing before we can build - we need to load the nconf config file from plan.sh - add this content:
+
+**plan.sh**
+
+```
+pkg_name=learn-about-me
+pkg_origin=core
+pkg_version="0.1.0"
+pkg_scaffolding="core/scaffolding-node"
+
+pkg_binds=(
+  [database]="port"
+)
+
+pkg_exports=(
+  [port]=app.port
+)
+
+declare -A scaffolding_env
+
+# Define path to config file
+scaffolding_env[APP_CONFIG]="{{pkg.svc_config_path}}/config.json"
+```
+
+Now, enter the Habitat studio with this command:
+
+```console
+$ hab studio enter
+```
+
+And, once you're in the studio, build the application:
+
+```console
+(studio) $ build
+```
+
+When the build is complete, export your new package as a Docker container image
+
+```console
+(studio) $ hab pkg export docker ./results/<your_new_package>.hart
+```
+
+And now run this command to pull down the core/mongodb package from the public Habitat Builder and export it as a Docker container image on your workstation:
+
+```console
+(studio) $ hab pkg export docker core/mongodb
+```
+
+Once that is complete, exit out of the studio
+
+```console
+(studio) $ exit
+```
+
+## Habitizing your application - Part III
+
+Now let's set up a Docker compose file using the two container images we just created - one will run the application itself, and one will run the MongoDB database.
+
+**docker-compose.yml**
+
+```
+version: '3'
+services:
+		mongodb:
+				image: core/mongodb
+				environment:
+				  # The default bind_ip for mongodb is to 127.0.0.1. Overriding
+				  # it here will allow our application to connect to through the
+			      # ip that is discovered through the sys.ip.
+				  HAB_MONGODB: "[mongod.net]\nbind_ip = '0.0.0.0'\n[mongod.security]\ncluster_auth_mode = ''"
+		learn-about-me-app:
+				image: core/learn-about-me
+				ports:
+				  - 8000:8000
+				# Find the container above named here by the peer
+				# Set the binding 'database' to the mongdb.default service
+				# group. Which the above mongodb is a member.
+				command: --peer mongodb --bind=database:mongodb.default
```

Comment:
I feel like this could use more explaining. It's a huge concept for those new to Habitat. Of course, if the focus of the blog post is to get it done leave it as is.